crates/telio-firewall/src/firewall.rs

```diff
@@ -244,26 +282,42 @@ pub struct StatefullFirewall {
     /// Wheter to still keep track of whitelisted TCP/UDP connections.
     /// Used for connection reset mechanism
     record_whitelisted: bool,
+    /// Local node ip addresses
+    ip_address: RwLock<Vec<StdIpAddr>>,
+    /// Capturing local interfaces struct
+    get_if_addr: G,
+    /// Custom IPv4 range to check against
+    custom_ip_range: Option<Ipv4Network>,
 }
 
-#[derive(Debug)]
-struct UdpConnectionInfo {
+#[derive(Debug, PartialEq)]
+struct ConnectionInfo {
     is_remote_initiated: bool,
+    connection_detail: ConnectionDetail,
+}
+
+#[derive(Debug, PartialEq)]
+enum ConnectionDetail {
```

Comment:
nitpick: this enum name could be `Transport` or `TransportProtocol` or something more explicit